poisson_lib.py

```
@@ -128,647 +166,629 @@ def analyze_poisson_outcome(
+        y = pd.to_numeric(df[outcome_name], errors='coerce').dropna()
+        if len(y) == 0:
+            return "<div class='alert'>⚠️ No valid data rows available.</div>", {}, {}, {}
+
+        df_aligned = df.loc[y.index]
+        total_n = len(y)
+        
+        # Handle offset if provided
+        offset = None
+        if offset_col and offset_col in df_aligned.columns:
+            offset = np.log(pd.to_numeric(df_aligned[offset_col], errors='coerce').fillna(1))
+            logger.info(f"Using offset column: {offset_col}")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)